gcc/rust/privacy/rust-privacy-reporter.cc

```diff
@@ -93,161 +96,218 @@ void
 PrivacyReporter::visit (HIR::IdentifierExpr &ident_expr)
 {}
 
-void
-PrivacyReporter::visit (HIR::Lifetime &lifetime)
-{}
-
-void
-PrivacyReporter::visit (HIR::LifetimeParam &lifetime_param)
-{}
-
 void
 PrivacyReporter::visit (HIR::PathInExpression &path)
 {
   check_for_privacy_violation (path.get_mappings ().get_nodeid (),
 			       path.get_locus ());
 }
 
-void
-PrivacyReporter::visit (HIR::TypePathSegment &segment)
-{}
-
-void
-PrivacyReporter::visit (HIR::TypePathSegmentGeneric &segment)
-{}
-
 void
 PrivacyReporter::visit (HIR::TypePathSegmentFunction &segment)
-{}
+{
+  // FIXME: Do we need to do anything for this?
```

Comment:
We haven't done anything for these yet in general i always have to lookup the rust reference to remember what they are.

Reply:
Mmh interesting. I'll look into it when figuring out the last bits of type privacy reporting